### PR TITLE
chore: add requirements.txt and review DEV_NOTES.md

### PR DESCRIPTION
**Pull request by marregui**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+setuptools>=45.2.0
+Cython>=0.29.24
+wheel>=0.34.2
+cibuildwheel>=2.8.0
+Sphinx>=5.0.2
+sphinx-rtd-theme>=1.0.0
+tox>=3.25.1
```
Comment (amunra) [inline requirements.txt]: I'm not sure this is correct:

If I understand correctly requirements is for runtime, whilst we only really care about having these tools (incidentally, tox is no longer required) at build time.

Probably worth explanding:

```
[project.optional-dependencies]
publish = ["twine", "wheel"]
ci = ["cibuildwheel"]
```

and

```
[build-system]
requires = [
    # Setuptools 18.0 and above properly handles Cython extensions.
    "setuptools>=45.2.0",
    "wheel>=0.34.2",
    "cython>=0.29.24"]
```

In `pyproject.toml`.

File: proj.py [hunk 2]
```diff
@@ -20,13 +20,12 @@ def _run(*args, env=None, cwd=None):
...
     """
     cwd = cwd or PROJ_ROOT
-    args = [str(arg) for arg in args]
     sys.stderr.write('[CMD] ')
     if env is not None:
         env_str = ' '.join(f'{k}={shlex.quote(v)}' for k, v in env.items())
         sys.stderr.write(f'{env_str} ')
         env = {**os.environ, **env}
-    escaped_cmd = ' '.join(shlex.quote(arg) for arg in args)
+    escaped_cmd = ' '.join(shlex.quote(str(arg)) for arg in args)
```
Comment (amunra) [inline proj.py:28]: Nice catch.

File: README.rst [hunk 3]
```diff
@@ -9,15 +9,6 @@ This client library implements QuestDB's variant of the
...
-Status of the library
-=====================
-
-Work in progress: do not use yet.
-
-There will be an announcement on our `slack <http://slack.questdb.io>`_ once
-it's ready for use.
-
-
 Installation
```
Comment (amunra) [inline README.rst:12]: Leave this in, for now.

File: DEV_NOTES.rst [hunk 4]
```diff
@@ -1,213 +0,0 @@
-=================
-Development Notes
-=================
```
Comment (amunra) [inline DEV_NOTES.rst:3]: Why are you deleting the file?

File: examples/line_sender.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+from questdb.ilp import Sender
+
+if __name__ == "__main__":
+    with Sender(host='localhost', port=9009) as sender:
```
Comment (amunra) [inline examples/line_sender.py:4]: This example, and more like this, including authentication and TLS, shoud be exposed to the docs.

File: examples/line_sender.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+            sender.row(
+                'tab',
+                symbols={
+                    'name_a': 'val_a'
+                },
+                columns={
+                    'name_b': True,
+                    'name_c': 42,
+                    'name_d': 2.5,
+                    'name_e': 'val_b'
```
Comment (amunra) [inline examples/line_sender.py:15]: Probably a separate example with also `at=` and a `datetime.datetime` timestamp.

**Review by amunra — CHANGES_REQUESTED** [review]

**Comment by marregui** [comment]
will open a new PR, this one serve the purpose of discussing the project etc.

**Pull request closed without merging**